frontends/web/src/routes/device/bitbox02/setup/wallet-restore.tsx

```diff
@@ -72,22 +72,34 @@ export const RestoreFromMnemonic = ({
   onAbort,
 }: Props) => {
   const { t } = useTranslation();
+  const [errorCode, setErrorCode] = useState<number>();
+  const inProgress = useRef<string>();
 
   useEffect(() => {
-    restoreFromMnemonic(deviceID)
-      .then(result => {
-        if (!result.success) {
-          const errorText = result.code === errUserAbort
-            ? t('bitbox02Wizard.restoreFromMnemonic.e104')
-            : t('bitbox02Wizard.restoreFromMnemonic.failed');
-          alertUser(errorText, {
-            asDialog: false,
-            callback: () => onAbort(),
-          });
-        }
-      })
-      .catch(console.error);
-  }, [deviceID, onAbort, t]);
+    if (inProgress.current !== deviceID) {
+      // remember for which device the workflow already has been initialized
+      inProgress.current = deviceID;
+      restoreFromMnemonic(deviceID)
+        .then(result => {
+          if (!result.success) {
+            setErrorCode(result.code);
+          }
+        })
+        .catch(console.error);
+    }
+  }, [deviceID]);
+
+  useEffect(() => {
```

Comment:
why a separate useEffect and not just handle it above like before?

Reply:
I noticed that onAbort callback changed and caused the effect with restoreFromMnemonic to execute again. So I tried to make an effect with as little depenencies as possible. and one for error handling.